netket/experimental/driver/vmc_kernel.py

```diff
@@ -0,0 +1,164 @@
+    the following parallel implementation is described in Appendix "Distributed SR computation".
+    """
+    N_params = O_L.shape[-1]
+    N_mc = O_L.shape[0] * n_nodes
+
+    if N_params % n_nodes != 0:
+        raise NotImplementedError() #* in this case O_L should be padded with zeros
+
+    O_L = O_L / N_mc**0.5
+    dv = -2.0 * de / N_mc**0.5
+
+    if mode=="complex":
+        O_L = jnp.concatenate((O_L[:, 0], O_L[:, 1]), axis=0)
+        dv = jnp.concatenate((jnp.real(dv), -jnp.imag(dv)), axis=-1)
+    elif mode=="real":
+        dv = dv.real
+    else:
+        raise NotImplementedError()
+
+    O_LT = rearrange(O_L, 'twons (np proc) -> proc twons np', proc=n_nodes)
+    
+    dv, token = mpi_gather_jax(dv)
+    dv = dv.reshape(-1, *dv.shape[2:])
+    O_LT, token = mpi_alltoall_jax(O_LT, token=token)
+
+    O_LT = rearrange(O_LT, 'proc twons np -> (proc twons) np')
+    
+    matrix, token = mpi_reduce_sum_jax(O_LT@O_LT.T, token=token)
+    matrix_side = matrix.shape[-1] #* it can be Ns or 2*Ns, depending on mode
+
+    if rank==0:
+        matrix = matrix + diag_shift * jnp.eye(matrix_side) #* shift diagonal regularization
+        aus_vector = solver_fn(matrix, dv)
+        aus_vector = aus_vector.reshape(n_nodes, -1)
+        aus_vector, token = mpi_scatter_jax(aus_vector, token=token)
+    else:
+        shape = jnp.zeros((int(matrix_side/n_nodes),), dtype=jnp.float64)
+        aus_vector, token = mpi_scatter_jax(shape, token=token)
+
+    updates = O_L.T @ aus_vector
+    updates, token = mpi_allreduce_sum_jax(updates, token=token)
+    
+    return -updates 
+
+inv_default_solver = lambda A, b: jnp.linalg.inv(A) @ b
+linear_solver = lambda A, b: jsp.linalg.solve(A, b, assume_a="pos")
+
+class VMC_kernelSR(VMC):
+    """
+    Energy minimization using Variational Monte Carlo (VMC) and the kernel 
+    formulation of Stochastic Reconfiguration (SR). This approach lead to
+    *exactly* the same parameter updates of the standard SR with a
+    diagonal shift regularization. For this reason, it is equivalent to the standard 
+    nk.driver.VMC with the preconditioner nk.optimizer.SR(solver=netket.optimizer.solver.solvers.solve))
+    In the kernel SR framework, the updates of the parameters can be written as:
+
+    :math:`\delta \\theta = \\tau X(X^TX + \lambda \mathbb{I}_{2M})^{-1} f`,
+
+    where :math:`X \in R^{P \\times 2M}` is the concatenation of the real and imaginary part 
+    of the centered Jacobian, with P the number of parameters and M the number of samples.
+    The vector f is the concatenation of the real and imaginary part of the centered local
+    energy. Note that, to compute the updates, it is sufficient to invert an :math:`M\\times M` matrix
+    instead of a :math:`P\\times P` one. As a consequence, this formulation is useful
+    in the typical deep learning regime where :math:`P \gg M`.
+    
+    See R.Rende, L.L.Viteritti, L.Bardone, F.Becca and S.Goldt (https://arxiv.org/abs/2310.05715)
```

Comment:
this should be an hyperlink (the syntax is terrible... I know),

```suggestion
    See `R.Rende, L.L.Viteritti, L.Bardone, F.Becca and S.Goldt <https://arxiv.org/abs/2310.05715>`_
```